### ISSUE TITLE
STATUS OF PROJECT: and some crash at loading

### ISSUE THREAD

**Issue by mckaygerhard** [Poikilos/EnlivenMinetest#613]
hi , i download the master sources and also the provided binaries in releases, and also the listed in minetest io domain.. 

none of them works with minetest releases since 0.4.16+

i try to made the buch of game fro this repo, but the process is TOOO comploicated: most important part it download bunch of things that are not integrated to system package, and later cause conflicts with system package upgrades

the game show this last error:

```
2023-08-14 13:52:09: INFO[Main]: Compiling high level shaders for nodes_shader
2023-08-14 13:52:09: INFO[Main]: getShaderIdDirect(): Returning id=4 for name "nodes_shader"
2023-08-14 13:52:09: INFO[Main]: SourceShaderCache::getOrLoad(): No path found for "nodes_shader/opengl_geometry.glsl"
2023-08-14 13:52:09: INFO[Main]: Compiling high level shaders for nodes_shader
2023-08-14 13:52:09: INFO[Main]: getShaderIdDirect(): Returning id=5 for name "nodes_shader"
minetest: /build/minetest-yzOa48/minetest-5.1.1+repack/src/nodedef.cpp:741: void ContentFeatures::updateTextures(ITextureSource*, IShaderSource*, irr::scene::IMeshManipulator*, Client*, const TextureSettings&): Assertion `liquid_type == LIQUID_FLOWING' failed.
Aborted (core dumped)
```

the initiator seems good, but i got that:

```
2023-08-14 13:59:38: VERBOSE[Main]: NodeDefManager: registering content id "21424": name="unifiedbricks:multicolor_medium_yellow"
2023-08-14 13:59:38: INFO[Main]: Starting server on 0.0.0.0...
        .__               __                   __   
  _____ |__| ____   _____/  |_  ____   _______/  |_ 
 /     \|  |/    \_/ __ \   __\/ __ \ /  ___/\   __\
|  Y Y  \  |   |  \  ___/|  | \  ___/ \___ \  |  |  
|__|_|  /__|___|  /\___  >__|  \___  >____  > |__|  
      \/        \/     \/          \/     \/        
2023-08-14 13:59:38: ACTION[Main]: World at [/home/general/.minetest/worlds/bucketgameworld]
2023-08-14 13:59:38: ACTION[Main]: Server for gameid="Bucket_Game" listening on 0.0.0.0:30000.
2023-08-14 13:59:38: INFO[Main]: SourceImageCache::getOrLoad(): Loading path "/usr/share/games/minetest/textures/base/pack/minimap_mask_round.png"
2023-08-14 13:59:38: INFO[Main]: SourceImageCache::getOrLoad(): Loading path "/usr/share/games/minetest/textures/base/pack/minimap_overlay_round.png"
2023-08-14 13:59:38: INFO[Main]: SourceImageCache::getOrLoad(): Loading path "/usr/share/games/minetest/textures/base/pack/minimap_mask_square.png"
2023-08-14 13:59:38: VERBOSE[Server]: registerCraft: registering craft definition: (shapeless, output="farming:porridge", recipe={ "group:food_barley","group:food_barley","group:food_wheat","group:food_wheat","group:food_bowl","group:food_milk" }, replacements={"mobs:bucket_milk"=>"bucket:buck
2023-08-14 13:59:38: VERBOSE[Server]: et_empty"})
2023-08-14 13:59:38: WARNING[Server]: WARNING: minetest.setting_* functions are deprecated.  Use methods on the minetest.settings object. (at ...test/games/Bucket_Game/mods/codercore/timespeed/init.lua:30)
2023-08-14 13:59:38: INFO[Server]: stack traceback:
2023-08-14 13:59:38: INFO[Server]: 	[C]: in function 'log'
2023-08-14 13:59:38: INFO[Server]: 	/usr/share/games/minetest/builtin/game/deprecated.lua:61: in function 'setting_set'
2023-08-14 13:59:38: INFO[Server]: 	...test/games/Bucket_Game/mods/codercore/timespeed/init.lua:30: in function <...test/games/Bucket_Game/mods/codercore/timespeed/init.lua:24>
2023-08-14 13:59:38: INFO[Server]: 	/usr/share/games/minetest/builtin/game/register.lua:429: in function </usr/share/games/minetest/builtin/game/register.lua:413>
2023-08-14 13:59:38: INFO[Main]: SourceImageCache::getOrLoad(): Loading path "/usr/share/games/minetest/textures/base/pack/minimap_overlay_square.png"
2023-08-14 13:59:38: INFO[Main]: SourceImageCache::getOrLoad(): Loading path "/usr/share/games/minetest/textures/base/pack/player_marker.png"
2023-08-14 13:59:38: INFO[Main]: SourceImageCache::getOrLoad(): Loading path "/usr/share/games/minetest/textures/base/pack/object_marker_red.png"
2023-08-14 13:59:38: INFO[Main]: Connecting to server at 127.0.0.1:30000
2023-08-14 13:59:38: INFO[Main]: Client packetcounter (20):
2023-08-14 13:59:38: VERBOSE[Server]: Server::peerAdded(): peer->id=2
2023-08-14 13:59:38: INFO[Main]: Client::peerAdded(): peer->id=1
2023-08-14 13:59:38: VERBOSE[Server]: Server: Handling peer change: id=2, timeout=0
2023-08-14 13:59:38: INFO[Server]: Server: Maximum lag peaked to 2.016 s

```

i f we take examples like alpine distros, that are so so simplistic that we can find the problem when something does not work in a docker.. 

i do not have idea where to start with this .. but i can help to report and track any problem but limited due nature of complications

in any case i have a bunch of good amount of mods ported to work with 0.4 and 5.X some mods needs work maybe,.

**Comment by Poikilos** [comment]
The scripts here are to help install packages necessary build. The dependency install script needs to be fixed due to python-dev being deprecated. So are you trying to run bucket_game in Minetest 5.1.1? I'm not sure that will work. 0.5 and 5.0 and on removed API functions. The copy of 5.3 in Linux Minetest Kit works, but it has the attached patch which includes some commits. I'm releasing it as per the LGPL v2.1 (see following URL for authors of commit we used). The program isn't released yet so I don't think we need to release code we haven't even given binaries for, but I will attach the patch anyway for your benefit and for everyone. The patch is based on 5.3.0-dev, this commit: https://github.com/minetest/minetest/commit/ee6c499ff8b9abf0d22ebc817a26f33322ffdd5b Some cherry picking may have been done to remove other commits, but I did `git diff master origin/master` to create the patch so it should work.
[trolltest-2023-08-04-vs-ee6c499.patch.zip](https://github.com/Poikilos/EnlivenMinetest/files/12340209/trolltest-2023-08-04-vs-ee6c499.patch.zip). If OldCoder has had you help with testing and given you linux-minetest-kit that will help but if you give out binaries please give the source code as well to remain compliant with LGPL v2.1.

**Comment by Poikilos** [comment]
> i try to made the buch of game fro this repo, but the process is TOOO comploicated: most important part it download bunch of things that are not integrated to system package, and later cause conflicts with system package upgrades

linux-minetest-kit should not cause conflicts with the system--it will not work with run-in-place because it installs its own modified libraries in some cases (which run from its own directory not the system). Did my scripts install a package that caused a problem?


> the game show this last error:
> 
> ```
> 2023-08-14 13:52:09: INFO[Main]: Compiling high level shaders for nodes_shader
> 2023-08-14 13:52:09: INFO[Main]: getShaderIdDirect(): Returning id=4 for name "nodes_shader"
> 2023-08-14 13:52:09: INFO[Main]: SourceShaderCache::getOrLoad(): No path found for "nodes_shader/opengl_geometry.glsl"
> 2023-08-14 13:52:09: INFO[Main]: Compiling high level shaders for nodes_shader
> 2023-08-14 13:52:09: INFO[Main]: getShaderIdDirect(): Returning id=5 for name "nodes_shader"
> minetest: /build/minetest-yzOa48/minetest-5.1.1+repack/src/nodedef.cpp:741: void ContentFeatures::updateTextures(ITextureSource*, IShaderSource*, irr::scene::IMeshManipulator*, Client*, const TextureSettings&): Assertion `liquid_type == LIQUID_FLOWING' failed.
> Aborted (core dumped)
> ```
> 
> the initiator seems good, but i got that:
> ...
> 
> ...
> 2023-08-14 13:59:38: ACTION[Main]: World at [/home/general/.minetest/worlds/bucketgameworld]


From the log your copy of Minetest doesn't look like a minetest.org binary because it is not run-in-place (uses ~/.minetest) and because it is 5.1.

Exactly which Minetest gave the error?

**Comment by mckaygerhard** [comment]
> Even OldCoder's modified copy of 5.3 will not run bucket_game so I recommend MultiCraft2 (older version) or Final Minetest binaries instead.

err @Poikilos  **this is so unconfortable, too exclusive and too exotic to play**

do you know that i have 3 subgames that runs in any engine version from 0.4.16 to 5.4 ? including multicraft ? and it runs in mobile also..

**Comment by mckaygerhard** [comment]
apart that i cannot packages final minetest for alpine linux that does not runs glibc.. really we need to make this exclusive way only after we have a huge base players, like mycrosoft made by example.. 

i check downloads and seems only for crappy windoser user are provide by the way

i really appreciated more flexible way to provide a door to help and work togetter.. please

**Comment by Poikilos** [comment]
> > Even OldCoder's modified copy of 5.3 will not run bucket_game so I recommend MultiCraft2 (older version) or Final Minetest binaries instead.
> 
> err @Poikilos **this is so unconfortable, too exclusive and too exotic to play**
> 
> do you know that i have 3 subgames that runs in any engine version from 0.4.16 to 5.4 ? including multicraft ? and it runs in mobile also..

IMO 5.0, 5.1, 5.2, 5.3, 5.4, 5.5, 5.6, and 5.7 are each too exotic and uncomfortable to use as an API target, even compared to the previous version in the list. There is no point. The number of incompatible mods is maybe 500, then multiplied by games those mods are in (copied and sometimes customized) is likely over 10,000, then multiplied by the number of servers running either upstream mod or original game is likely well over a million. I have described it at these two locations:

- https://minetest.io
- https://poikilos.org/2020/05/09/minetest-scope-issue/

There is no longer a way to fix those one million copies of mods even if upstream ones are made compatible with MT5. It is a tragedy and can't be erased except by ignoring the MT5 client and server and giving out a different one.

I would be glad to help you make mods work with older clients and already have forks on GitHub that do so and OldCoder has done so in copies of them in bucket_game.

However, making a million things work with MT5 is letting the engine designers make themselves more important than the ones who make the games. As the MT5 designers continue this, they create a narcissistic and destructive and a waste of hundreds of hours times hundreds of people, possibly up to hundreds of thousands of combined hours.

**Comment by mckaygerhard** [comment]
> > > Even OldCoder's modified copy of 5.3 will not run bucket_game so I recommend MultiCraft2 (older version) or Final Minetest binaries instead.
> > err @Poikilos **this is so unconfortable, too exclusive and too exotic to play**
> IMO 5.0, 5.1, 5.2, 5.3, 5.4, 5.5, 5.6, and 5.7 are each too exotic and uncomfortable t...

all my games and mods currently works with 0.4.16 and those version you cited.. so ITS NOT IMPOSSIBLE

> There is no longer a way to fix those one million copies of mods even if upstream ones are made compatible with MT5. It is a tragedy and can't be erased except by ignoring the MT5 client and server and giving out a different one.
> 

i already made it, if you and me works togetter will made it better.. but currently this repository its not understandable for normal player (mckayshirou cited me such thing strongly)

> I would be glad to help you make mods work with older clients and already have forks on GitHub that do so and OldCoder has done so in copies of them in bucket_game.

again i already do it: https://git.minetest.io/minenux/

> 
> However, making a million things work with MT5 is letting the engine designers make themselves more important than the ones who make the games. As the MT5 designers continue this, they create a narcissistic and destructive waste of hundreds of hours times hundreds of people, possibly up to hundreds of thousands of combined hours.

**Comment by Poikilos** [comment]
Then what are you trying to do? You haven't answered. You say there is a crash but you don't say what you want. Whose minetest do you want to run? Do you want bucket_game to run on it?

**Comment by Poikilos** [comment]
You said also,
> sources and also the provided binaries in releases, and also the listed in minetest io domain..
> 
> none of them works with minetest releases since 0.4.16+

none of what?

What are you talking about?

Tell me what you want to do.

If a binary didn't work, tell me which one and what didn't work, and what didn't work with it.

**Comment by Poikilos** [comment]
If we talk on Discord I think we should discuss more there since you want to discuss things not just solve an issue. You mentioned several things unrelated to your log. Your log indicates you want to run bucket_game on Minetest 5.1.x and OldCoder indicates bucket_game has checks to make it work on MT 5 to the extent that it is convenient. If you tell me where the binaries or source are for what you have, that would help. I don't know where to get a working MT5.1 right now. The AppImages were sandboxed incorrectly and don't run due to library conflicts which is exactly what AppImage was designed to avoid. It is more broken amature stuff from MT5. I disagree you have fixed everything because games like bucket game and others times servers means millions are still left out of MT5's ecosystem. You have created a new ecosystem and that's very good. Others are still left out but you have helped a lot. Please chat with me on voice so I can better understand your goals and if there is some way we can work together. Discord.io suffered a data breach but it doesn't appear to affect the safety of my Discord invites.

- The details if the breach are here if you are concerned: https://discord.io
- It seems to be working even though the main page discord.io says it is down, but if you have trouble with the invite link on https://zahyest.com/, I can give you a personal invite that doesn't use their site (recommended for the most safety since they are dealing with unknowns): Add me at `@poikilos_` on Discord.

**Comment by Poikilos** [comment]
> > I would be glad to help you make mods work with older clients and already have forks on GitHub that do so and OldCoder has done so in copies of them in bucket_game.
> 
> again i already do it: https://git.minetest.io/minenux/
> 
> > However, making a million things work with MT5 is letting the engine designers make themselves more important than the ones who make the games. As the MT5 designers continue this, they create a narcissistic and destructive waste of hundreds of hours times hundreds of people, possibly up to hundreds of thousands of combined hours.

Also, it breaks again and again, not just in MT5, and not just today for things you fixed. They break again and again so you get nowhere. They are making everyone tread water and not make progress so they feel important. You did all that work, but just in one ecosystem...and even that one will break with 5.2, again in 5.3, again in 5.4, again in 5.5, again in 5.6, again in 5.7, and it makes me nausious. The engine should stay out of the way, not assert itself and break thousands of hours of work. It is not "evolution" it is arbitrary nonsense to change API calls. Evolution isn't infinitely upward, it is adaptation. People are deluded into thinking they will become gods. I don't worship these deranged people.

**Comment by mckaygerhard** [comment]
hi .. ok, i already tested with, vanilla 5.1.1 !

**Can you provide a downloable minetest bucket game and eliven minetest game in the pages? **

### About the subgames:

can you provide a working dowload of eliven minetest in minetst.org and minetest.io ? thanks in advance

i used the vanilla 5.1 engine and same result, what i can do to help to make **it work? and can we provided a working relesae in the minetest.org and minetest.io pages?**

log of bucket:

```
2023-08-14 13:59:38: ACTION[Main]: World at [/home/general/.minetest/worlds/bucketgameworld]
2023-08-14 13:59:38: ACTION[Main]: Server for gameid="Bucket_Game" listening on 0.0.0.0:30000.
2023-08-14 13:59:38: INFO[Main]: SourceImageCache::getOrLoad(): Loading path "/usr/share/games/minetest/textures/base/pack/minimap_mask_round.png"
2023-08-14 13:59:38: INFO[Main]: SourceImageCache::getOrLoad(): Loading path "/usr/share/games/minetest/textures/base/pack/minimap_overlay_round.png"
2023-08-14 13:59:38: INFO[Main]: SourceImageCache::getOrLoad(): Loading path "/usr/share/games/minetest/textures/base/pack/minimap_mask_square.png"
2023-08-14 13:59:38: VERBOSE[Server]: registerCraft: registering craft definition: (shapeless, output="farming:porridge", recipe={ "group:food_barley","group:food_barley","group:food_wheat","group:food_wheat","group:food_bowl","group:food_milk" }, replacements={"mobs:bucket_milk"=>"bucket:buck
2023-08-14 13:59:38: VERBOSE[Server]: et_empty"})
2023-08-14 13:59:38: WARNING[Server]: WARNING: minetest.setting_* functions are deprecated.  Use methods on the minetest.settings object. (at ...test/games/bucket/mods/codercore/timespeed/init.lua:30)
2023-08-14 13:59:38: INFO[Server]: stack traceback:
2023-08-14 13:59:38: INFO[Server]: 	[C]: in function 'log'
2023-08-14 13:59:38: INFO[Server]: 	/usr/share/games/minetest/builtin/game/deprecated.lua:61: in function 'setting_set'
2023-08-14 13:59:38: INFO[Server]: 	...test/games/bucket/mods/codercore/timespeed/init.lua:30: in function <...test/games/bucket/mods/codercore/timespeed/init.lua:24>
2023-08-14 13:59:38: INFO[Server]: 	/usr/share/games/minetest/builtin/game/register.lua:429: in function </usr/share/games/minetest/builtin/game/register.lua:413>
2023-08-14 13:59:38: INFO[Main]: SourceImageCache::getOrLoad(): Loading path "/usr/share/games/minetest/textures/base/pack/minimap_overlay_square.png"
2023-08-14 13:59:38: INFO[Main]: SourceImageCache::getOrLoad(): Loading path "/usr/share/games/minetest/textures/base/pack/player_marker.png"
2023-08-14 13:59:38: INFO[Main]: SourceImageCache::getOrLoad(): Loading path "/usr/share/games/minetest/textures/base/pack/object_marker_red.png"
2023-08-14 13:59:38: INFO[Main]: Connecting to server at 127.0.0.1:30000
2023-08-14 13:59:38: INFO[Main]: Client packetcounter (20):
2023-08-14 13:59:38: VERBOSE[Server]: Server::peerAdded(): peer->id=2
2023-08-14 13:59:38: INFO[Main]: Client::peerAdded(): peer->id=1
2023-08-14 13:59:38: VERBOSE[Server]: Server: Handling peer change: id=2, timeout=0
2023-08-14 13:59:38: INFO[Server]: Server: Maximum lag peaked to 2.016 s
<here just hang with no error>
```

i downloaded the **ELIVEN from releases here, and from git, the RELESE 0.4.15 works with 0.4.17, but not with 5.1,** the git sources does not work in any version

### About mods and breaking

you have right with "break" again and again.. i will try to made compatibles with any engine as posible as a form of help to all of us

> **You need the latest version of Bucket Game. It should work with MT 5.1.

thanks in advance...

**Comment by Poikilos** [comment]
For years I cannot find any version of vines that works with any version of minetest even 5.x. Either it crashes or I fix the crash and no vines generate. so if you can get me a vines mod that works with anything and tell me what it works with then I could fix it in ENLIVEN and put other updated mods in it for a release. Vines are necessary for ropes and other things.

bucket_game is up to OldCoder to release. He has offered you a prerelease for servers but he said he is not ready to release it yet (but is close).

**Comment by Poikilos** [comment]
The release of the latest bucket_game with Minetest 5.3 support is now available on [minetest.org](https://minetest.org). However, wait for the release with issue #615 fixed.

**Comment by mckaygerhard** [comment]
## QUICK TESTS:

I i report feedback, unfortunately due the hour and at the end of the weekend i cannot get too deep ...

but i will **try to check it more deep in the course of the week for some corrections to help them run in any environment** without depending on a version.. I REPEAT: minenux mods run both in 0.4 and 5.4, in any version even in multicraft without limits

#### minetest org  5.4 compilation with patches backported

* linux debian 12 lasted , DELL optiplez 7071 cpu  i9 9990, gpu radeon RX 550, 4G ram

```
2023-08-20 20:14:02: ERROR[Main]: ModError: Failed to load and run script from /home/general/.minetest5/games/bucket_game/mods/upper_realms/asteroid/init.lua:
2023-08-20 20:14:02: ERROR[Main]: Unknown node: birthstones:emeraldore
2023-08-20 20:14:02: ERROR[Main]: stack traceback:
2023-08-20 20:14:02: ERROR[Main]: 	[C]: in function 'get_content_id'
2023-08-20 20:14:02: ERROR[Main]: 	...t5/games/bucket_game/mods/upper_realms/asteroid/init.lua:153: in main chunk
2023-08-20 20:14:02: ERROR[Main]: Revisa debug.txt para más detalles.
2023-08-20 20:14:02: ACTION[Main]: Server: Shutting down
```

#### multicraft lasted 

* linux debian 7 wheeze , cpu  i7, gpu radeon hd 7570, 1G ram

```
2023-08-20 19:59:13: ACTION[Main]: max_homes is 8
2023-08-20 19:59:13: ERROR[Main]: ModError: Failed to load and run script from /home/general/.multicraft/games/bucket_game/mods/codercore/ocsethome/init.lua:
2023-08-20 19:59:13: ERROR[Main]: ...icraft/games/bucket_game/mods/codercore/ocmkdir/init.lua:21: attempt to call field 'execute' (a nil value)
2023-08-20 19:59:13: ERROR[Main]: stack traceback:
2023-08-20 19:59:13: ERROR[Main]: 	...icraft/games/bucket_game/mods/codercore/ocmkdir/init.lua:21: in function 'mkdir'
2023-08-20 19:59:13: ERROR[Main]: 	...raft/games/bucket_game/mods/codercore/ocsethome/init.lua:36: in main chunk
2023-08-20 19:59:13: ERROR[Main]: Check debug.txt for details.
2023-08-20 19:59:13: ACTION[Main]: Server: Shutting down
```

#### minetest 4 0.4.17.1 with backported fixed from minenux project

* compiled agains alpine linux, same structued, Semprom cpu 1800MHz, openchrome gpu 4Mb gram (VIA), 1GB RAM

```
Loaded texture: /home/general/Minetest/bucket_game/menu/header.png
2023-08-20 20:27:51: WARNING[Main]: BanManager: creating /home/general/.minetest/worlds/bg-mt4/ipban.txt
2023-08-20 20:27:51: ERROR[Main]: ModError: Failed to load and run script from /home/general/.minetest/games/bucket_game/mods/underground/modlib/init.lua:
2023-08-20 20:27:51: ERROR[Main]: ...netest/games/bucket_game/mods/underground/modlib/log.lua:59: attempt to call field 'register_on_mods_loaded' (a nil value)
2023-08-20 20:27:51: ERROR[Main]: stack traceback:
2023-08-20 20:27:51: ERROR[Main]: 	...netest/games/bucket_game/mods/underground/modlib/log.lua:59: in function 'file'
2023-08-20 20:27:51: ERROR[Main]: 	...etest/games/bucket_game/mods/underground/modlib/init.lua:56: in function 'loadfile_exports'
2023-08-20 20:27:51: ERROR[Main]: 	...etest/games/bucket_game/mods/underground/modlib/init.lua:103: in main chunk
2023-08-20 20:27:51: ERROR[Main]: Revisa debug.txt para más detalles.
Loaded texture: /home/general/Minetest/bucket_game/menu/header.png
```

#### minetest org older bin 5.1.1 original debian package

* Debian 9 strech , i3 cpu, intel 915 gpu, 2G ram

```
2023-08-20 20:14:02: ERROR[Main]: ModError: Failed to load and run script from /home/general/.minetest5/games/bucket_game/mods/upper_realms/asteroid/init.lua:
2023-08-20 20:14:02: ERROR[Main]: Unknown node: birthstones:emeraldore
2023-08-20 20:14:02: ERROR[Main]: stack traceback:
2023-08-20 20:14:02: ERROR[Main]: 	[C]: in function 'get_content_id'
2023-08-20 20:14:02: ERROR[Main]: 	...t5/games/bucket_game/mods/upper_realms/asteroid/init.lua:153: in main chunk
2023-08-20 20:14:02: ERROR[Main]: Revisa debug.txt para más detalles.
2023-08-20 20:14:02: ACTION[Main]: Server: Shutting down
```

**Comment by Poikilos** [comment]
Thank you for specifying the game versions and what you are using. We are aware of `2023-08-20 19:59:13: ERROR[Main]: 	...icraft/games/bucket_game/mods/codercore/ocmkdir/init.lua:21: in function 'mkdir'` See #615 . I will look into the others.

**Comment by mckaygerhard** [comment]
> Thank you for specifying the game versions and what you are using. We are aware of `2023-08-20 19:59:13: ERROR[Main]: ...icraft/games/bucket_game/mods/codercore/ocmkdir/init.lua:21: in function 'mkdir'` See #615 . I will look into the others.

i already check that.. but i noted that this is not the only error also i discourage the invasive things that some mods do.. as quick appreciation of the mods

**i will try to be in sync with older issues already registered here event open new ones..** for the fixeds..

**Comment by Poikilos** [comment]
2023-08-20 20:14:02: ERROR[Main]: Unknown node: birthstones:emeraldore
is a separate issue, and sounds a lot like depends.txt instead of mod.conf is being used by mods in bucket_game. That is likely. We don't intend to make every mod work with every version. That is impossible no matter how much you say, because there are a million copies. bucket_game is just one of the examples of hundreds of games. Each of these errors are separate issues. I've already explained the status of the project. Please post a separate issue for separate issues, as that error is obviously separate. I already offered to chat. E-mail instead of making this issue unclear.

I've explained that we have different goals:

>  I REPEAT: minenux mods run both in 0.4 and 5.4, in any version even in multicraft without limits

I am happy for you but I am less happy you want to argue your point when I've already stated that your ecosystem is only yours and there are many mods and millions of copies of them. Though your ecosystem seems better than ContentDB and I'm happy for you, you are becoming like you just want to advertise and persuade me. You want to change every mod to work with every version of Minetest.

There is no reason to support Minetest 5.4. 5.5 will break your work, that's already been discussed. You say our builds are to "exotic" and "uncomfortable" and I've said that is true for each version of 5.x compared to the previous. Are you going to say using a XUL build of Mozilla is exotic and uncomfortable? Some people would rather use it so add-ons don't break every release and drain developers of time. Are you going to say that UPBGE is exotic and uncomfortable? Some people rather use it because Blender has become a walled garden where only the core team's supported add-ons get maintained so they don't break with each new release. Some people rather keep the features and add-ons that suit their purposes.

We will fix issues like #615 as it is convenient, but our goal is:

We want to make Minetest work with more mods, so millions of copies of hundreds of mods are not lost.

Please do not keep saying you "already made it". Think and speak clearly. You've made your own ecosystem and that's good but it doesn't change the millions of copies that aren't using your ecosystem. Please stop repeating yourself so I don't have to keep repeating myself. Understand and re-read my direct reply to your statement. Re-read. I don't like to repeat myself.

Be clear in your thinking and talking. You've brought up about 4 issues here, not 1. Stay on topic or make another issue. If you want to to talk about our philosophies, e-mail or chat.
- We can continue discussing here why my answer is the same as to our goals of supporting mods with the game rather than changing every mod,
- but you keep posting separate issues that prove my point and that you expect me to change every mod so I will lock this issue, as this is not a discussion board. I already locked it then unlocked it once but you've continue to post new issues that are separate. Your last post contained entirely different errors. You've also said you're using a minetest.org build of 5.4 but we don't have a build of 5.4, only 5.3-dev. Email or chat if you want to advertise your idea or learn how we can work together.

Please:
- for each of these errors post a separate issue not here. As for 615, check whether your issue already exists before posting here or on a new issue.
- Re-read and understand my answers above.
- Persuade or promote your mod ecosystem on a discussion board or e-mail or chat not on an issue.
